hal/src/nRF52840/exflash_hal.c

```diff
@@ -574,3 +581,33 @@ int hal_exflash_special_command(hal_exflash_special_sector_t sp, hal_exflash_com
     hal_exflash_unlock();
     return ret;
 }
+
+int hal_exflash_sleep(bool sleep, void* reserved) {
+    hal_exflash_lock();
+    if (sleep) {
+        // Suspend I2C
+        if (qspi_state != HAL_EXFLASH_STATE_ENABLED) {
+            hal_exflash_unlock();
+            return SYSTEM_ERROR_INVALID_STATE;
+        }
+        // Put external flash into sleep and disable QSPI peripheral
+        if (hal_exflash_special_command(HAL_EXFLASH_SPECIAL_SECTOR_NONE, HAL_EXFLASH_COMMAND_SLEEP, NULL, NULL, 0) != SYSTEM_ERROR_NONE ||
```

Comment:
1. This is probably going to fail if there is an ongoing write/erase operation. See the `HAL_EXFLASH_COMMAND_SUSPEND_PGMERS` logic in `hal_exflash_init`. Probably same thing should be done here.
2. It's not clear whether both `hal_exflash_special_command` and `hal_exflash_uninit` calls should happen. Please refactor to state the intent clearer, as right now it seems we are calling `hal_exflash_uninit` only if sleep command fails. If the sleep command does not fail, do we need to reinitialize pins? As far as I remember there is a leak through one of them.